### PR TITLE
[Merged by Bors] - Address Clippy 1.73 lints

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

Fix Clippy lints enabled by default in Rust 1.73.0, released today.

### CODE REVIEW

File: beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs [hunk 1]
```diff
@@ -330,13 +330,15 @@ impl Eq for Score {}
...
+        Some(self.cmp(other))
     }
 }
 
 impl Ord for Score {
     fn cmp(&self, other: &Score) -> std::cmp::Ordering {
-        self.partial_cmp(other).unwrap_or(std::cmp::Ordering::Equal)
+        self.score()
+            .partial_cmp(&other.score())
+            .unwrap_or(std::cmp::Ordering::Equal)
```
Comment (michaelsproul) [inline beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs:341]: Clippy thought the old impl was pretty [_suspicious_](https://rust-lang.github.io/rust-clippy/master/index.html#/incorrect_partial_ord_impl_on_ord_type).

The effect of Clippy's suggested change is that `partial_ord` will never return `None` for incomparable values (like `NaN`). I think this is OK, as we shouldn't be encountering this anyway (fingers crossed).

**Review by jimmygchen — APPROVED** [review]
Looks good! I've looked into the clippy suggestions as well, I think this change looks fine.

**Comment by jimmygchen** [comment]
Thanks!

bors r+

File: beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs (hunk 1 above)
Comment (jimmygchen) [reply beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs:341]: It doesn't look like there are any other scenarios (other than comparing `NaN`)  where `f64::partial_cmp` would return None, so this is probably fine 🤞

**Pull request closed without merging**